test/GeneratorTests.cpp

```diff
@@ -0,0 +1,25 @@
+#include <gtest/gtest.h>
+#include "Generator.hpp"
+#include <set>
+#include <cmath>
+
+struct GeneratorTests : public ::testing::Test
+{
+};
+
+TEST_F(GeneratorTests, checkAllPermutationOfSingleVector)
+{
+    auto result = getAllVectorPermutation_(3);
+    std::set<Line> s(result.begin(), result.end());
+    ASSERT_EQ(s.size(), pow(2,3));
+}
+
+TEST_F(GeneratorTests, checkConversionFromIntToFieldsEnum)
+{
+    Lines test = {{1, 1, 0},
```

Comment:
The name `test` is not good. What about `input`?

Reply:
Ok